pkg/apis/psmdb/v1/psmdb_types.go

```diff
@@ -839,6 +840,20 @@ type SecretsSpec struct {
 	LDAPSecret    string `json:"ldapSecret,omitempty"`
 }
 
+type VaultSpec struct {
+	EndpointURL string `json:"endpointURL,omitempty"`
```

Comment:
Should we set a required tag here? If yes, please ensure that the `VaultSpec` is set as a pointer type (i.e, `*VaultSpec`)